book/src/proposals/one-round-finality.md

```diff
@@ -0,0 +1,116 @@
+# Practical Finality in One Round of Consensus
+
+The cluster can achieve a high degree of finality though Tower
+consensus as the cluster commits to an exponentially increasing of
+safety for a block.  This safety is based on the commitment of any
+voting validator from switching votes to an alternative fork.  It
+is possible to achieve finality a different way, leaders can agree
+not to propose alternative forks.
+
+If only a part of the cluster has voted, then the set of all the
+leaders that haven't voted and the leader schedule can be used to
+simulate how lockout will grow in the future.  This calculation is
+based on the assumption that the partition is static, and that the
+same leaders will continue voting together.
+
+## Overview
+
+Leaders commit to producing a block that chains to the leaders last
+vote. Breaking this commitment is slashable.  The leader is free
+to switch forks and vote for an alternative chain.
+
+Once a block has been voted on, the subsequent leader schedule will
+reflect the likelihood of the block being finalized.
+
+For example:
+* 80% of the cluster voted for the block
+
+* The schedule for the next 5 leaders looks like:
+
+    * V,V,V, V,V,V,V, N,N,N,N, V,V,V,V, V,V,V,V
+
+Where V indicates that the leader voted on the block and N indicates
+that they haven't. By the time the `N` leader is reached, its likely
+this block would be voted on 7 times, and the lockout for this block
+would prevent the `N` leader from proposing an alternative fork
```

Comment:
Why hasn't `N` been voting on the block? Is it because they haven't observed it? If so and they haven't voted on the block, why would they be slashed for proposing an alternative fork?

Reply:
N is in a partition, and they are not slashed for proposing an alternative fork.  Only the nodes that voted for the block can be slashed for proposing fork that doesn't contain it.